execution/inc/leatherman/execution/execution.hpp

```diff
@@ -187,6 +187,16 @@ namespace leatherman { namespace execution {
     };
 
     /**
+     * Encapsulates return value from executing a process: whether or not the command succeeded, output from stdout, output from stderr (if not redirected), and the exit code.
+     */
+    struct exec_result
+    {
+        bool success;
```

Comment:
I prefer to have a default constructor when there are members that may not be default initialized.